tizen-api/include/tizen-api.h

```diff
@@ -0,0 +1,285 @@
+ * @param[in] buf The contents of the tensor output (a single frame. tensor/tensors).
+ * @param[in] size The size of the buffer.
+ * @param[in] tensorinfo The cardinality, dimension, and type of given tensor/tensors.
+ * @param[in/out] pdata User Application's Private Data
+ */
+typedef void (*nns_sink_cb)
+(const char *buf, size_t size, const GstTensorsInfo *tensorinfo, void *pdata);
+
+/**
+ * @brief Register a callback for sink (tensor_sink) of nnstreamer pipelines.
+ * @since_tizen 5.5
+ * @param[in] pipe The pipeline to be attached with a sink node.
+ * @param[in] sinkname The name of sink node, described with nns_pipeline_construct().
+ * @param[in] cb The function to be called by the sink node.
+ * @return The nns_sink_handle. Returns NULL if fails.
+ */
+nns_sink_handle nns_pipeline_sink_register
+(nns_pipeline_h *pipe, const char *sinkname, nns_sink_cb cb);
+
+/**
+ * @brief Unregister a callback for sink (tensor_sink) of nnstreamer pipelines.
+ * @since_tizen 5.5
+ * @param[in] The nns_sink_handle to be unregistered (destroyed)
+ * @return @c 0 on success. otherwise a negative error value
+ * @retval #NNS_ERROR_NONE Successful
+ */
+int nns_pipeline_sink_unregister (nns_sink_handle h)
+
+/**
+ * @brief Get a handle to operate as a src node of nnstreamer pipelines.
+ * @since_tizen 5.5
+ * @param[in] pipe The pipeline to be attached with a src node.
+ * @param[in] srcname The name of src node, described with nns_pipeline_construct().
+ * @param[out] tensorinfo The cardinality, dimension, and type of given tensor/tensors.
+ * @return The nns_src_handle. Returns NULL if fails.
+ */
+nns_src_handle nns_pipeline_src_gethandle
+(nns_pipeline_h *pipe, const char *srcname, GstTensorsInfo *tensorinfo);
+
+/**
+ * @brief Register a callback
+ * @param[out] bufpolicy The policy of buf deallocation.
+ * @return 0 on success (buf is filled). otherwise a negative error value.
+ * @retval #NNS_ERROR_NONE Successful
+ */
+int nns_pipeline_src_inputdata (nns_src_handle *handle,
+    nns_buf_policy policy, char *buf, size_t size)
+
+/****************************************************
+ ** NNStreamer Pipeline Switch/Valve Control       **
+ ****************************************************/
+
+/**
+ * @brief Get a handle to operate a "GstInputSelector / GstOutputSelector" node of nnstreamer pipelines
+ * @detail Refer to gstreamer.freedesktop.org/data/doc/gstreamer/head/gst-plugins-bad-plugins/html/gst-plugins-bad-plugins-input-selector.html for input selectors
+ *         Refer to https://gstreamer.freedesktop.org/data/doc/gstreamer/head/gstreamer-plugins/html/gstreamer-plugins-output-selector.html for output selectors
+ * @param[in] pipe The pipeline to be managed.
+ * @param[in] switchname The name of switch (InputSelector/OutputSelector)
+ * @param[out] num_nodes The number of nodes selected by this switch node. We assume they are listed from 0 ... num_nodes-1.
+ * @param[in] is_inputselector 0 if it's OutputSelector. 1 if it's InputSelector.
```

Comment:
We can also define a enum type for indicating the role of this switch.

Reply:
Updated